### PR TITLE
Some build change

### PR DESCRIPTION
**Pull request by zasdfgbnm**
- Use `ninja` instead of make, because `ninja` provides better parallelization and it will display the command when failed on compiling a file (and this is super helpful for debugging)
- Generate `compile_commands.json`
- ~Allow using system gtest and gbench (just trying to build less file)~

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
This is likely to conflict with https://github.com/NVIDIA/Fuser/pull/109, I will wait https://github.com/NVIDIA/Fuser/pull/109 to be merged and rebase.

**Comment by zasdfgbnm** [comment]
This is why I love `ninja`:

```C++
[4/10] Building CXX object CMakeFiles/nvfuser_codegen.dir/csrc/ir_nodes.cpp.o
FAILED: CMakeFiles/nvfuser_codegen.dir/csrc/ir_nodes.cpp.o 
/usr/bin/g++-10 -DTORCH_CUDA_BUILD_MAIN_LIB -DUSE_C10D_GLOO -DUSE_C10D_NCCL -DUSE_DISTRIBUTED -DUSE_RPC -DUSE_TENSORPIPE -Dnvfuser_codegen_EXPORTS -I/home/gaoxiang/Fuser/third_party/gloo -I/home/gaoxiang/Fuser/csrc -I/home/gaoxiang/Fuser/build/include -isystem /home/gaoxiang/cuda-11.8/include -isystem /home/gaoxiang/.local/lib/python3.10/site-packages/torch/include -isystem /home/gaoxiang/.local/lib/python3.10/site-packages/torch/include/torch/csrc/api/include -D_GLIBCXX_USE_CXX11_ABI=0 -std=gnu++17 -fPIC -Wall -Wno-unused-function -DTORCH_CUDA_BUILD_MAIN_LIB -Werror -D_GLIBCXX_USE_CXX11_ABI=0 -MD -MT CMakeFiles/nvfuser_codegen.dir/csrc/ir_nodes.cpp.o -MF CMakeFiles/nvfuser_codegen.dir/csrc/ir_nodes.cpp.o.d -o CMakeFiles/nvfuser_codegen.dir/csrc/ir_nodes.cpp.o -c /home/gaoxiang/Fuser/csrc/ir_nodes.cpp
/home/gaoxiang/Fuser/csrc/ir_nodes.cpp:1685:2: error: ‘fasdfasfasdfa’ does not name a type
 1685 | }fasdfasfasdfa
      |  ^~~~~~~~~~~~~
```
really appreciate the command it provides

**Review by jjsjann123 — APPROVED** [review]
:clap:

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,9 +24,19 @@ if (PROJECT_IS_TOP_LEVEL)
...
+    if(SYSTEM_GTEST)
+      find_package(GTest REQUIRED)
+    else()
+      add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/googletest)
+    endif()
+  endif()
   if(BUILD_NVFUSER_BENCHMARK)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/googletest)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/benchmark)
+    if(SYSTEM_GBENCH)
```
Comment (rdspring1) [inline CMakeLists.txt:35]: Partial support for this exists here: https://github.com/NVIDIA/Fuser/blob/main/Dependencies.cmake#L33

Comment (zasdfgbnm) [reply CMakeLists.txt:35]: WOW, thanks for doing that!

**Comment by zasdfgbnm** [comment]
Looks like my system's gtest has different ABI from PyTorch.... So I can not test if my build change work... I removed this part from my PR.

**Pull request merged**